src/anomalib/deploy/inferencers/torch_inferencer.py

```diff
@@ -128,21 +130,39 @@ def post_process(self, predictions: Tensor, metadata: dict | DictConfig | None =
         if metadata is None:
             metadata = self.metadata
 
+        # Some models return a Tensor while others return a list or dictionary. Handle both cases.
+        # TODO: This is a temporary fix. We will wrap this post-processing stage within the model's forward pass.
+
+        # Case I: Predictions could be a tensor.
         if isinstance(predictions, Tensor):
             anomaly_map = predictions.detach().cpu().numpy()
             pred_score = anomaly_map.reshape(-1).max()
-        else:
-            # NOTE: Patchcore `forward`` returns heatmap and score.
-            #   We need to add the following check to ensure the variables
-            #   are properly assigned. Without this check, the code
-            #   throws an error regarding type mismatch torch vs np.
+
+        # Case II: Predictions could be a dictionary of tensors.
+        elif isinstance(predictions, dict):
+            if "anomaly_map" in predictions:
+                anomaly_map = predictions["anomaly_map"].detach().cpu().numpy()
+            else:
+                raise KeyError("``anomaly_map`` not found in the predictions.")
```

Comment:
Efficient AD will also need to be fixed to work with this.
https://github.com/openvinotoolkit/anomalib/blob/2a04129ad146603be3b29ff66c749d38630f4b7b/src/anomalib/models/efficient_ad/torch_model.py#L351
I think calling `anomaly_map_combined` just `anomaly_map` would be okay.